BankAccounts/src/com/company/AccountService.java

```diff
@@ -0,0 +1,47 @@
+package com.company;
+
+import java.io.FileWriter;
+
+public class AccountService {
+
+    public void transfer(Account accountA, Account accountB, double amount) {
+        synchronized(accountA)
+        {
+            withdraw(accountA, amount);
+            System.out.print(amount+" is withdrawn from account a\n");
+
+            try{Thread.sleep(500);}
+            catch(Exception e){System.out.println(e);}
+
+            synchronized(accountB)
+            {
+                deposit(accountB, amount);
+                System.out.print(amount+" is deposited into account b\n");
+            }
+        }
+    }
+
+    private double deposit(Account account, double amount) {
+        try{
+            account.setBalance(account.getBalance() + amount);
+            FileWriter myWriter = new FileWriter(account.getAccountName() + ".txt");
+            myWriter.write(String.valueOf(account.getBalance()));
+            Thread.sleep(100);
+        } catch (Exception e) {
+            System.out.println("Something went wrong");
+        }
+        return account.getBalance();
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

Similar code fragments were detected in the same file at the following lines: 25:33, 37:45.  
Refactoring can help improve code maintainability. Consider reducing duplicate code by extracting it into a separate method. You can then replace duplicated code with calls to this new method.